lib/rouge/lexers/augeas.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Augeas < RegexLexer
+      title "Augeas"
+      desc "The Augeas programming language (augeas.net)"
+
+      tag 'augeas'
+      aliases 'aug'
+      filenames '*.aug'
+      mimetypes 'text/x-augeas'
+
+      def self.reserved
+        @reserved ||= Set.new %w(
+          _ let del store value counter seq key label autoload
+          incl excl transform test get put in after set
+        )
+      end
+
+      def self.types
+        @types ||= Set.new %w(
+          unit string regexp lens tree filter
+        )
+      end
+
+      state :basic do
+        rule %r/\s+/m, Text
+        rule %r/\(\*/, Comment::Multiline, :comment
+      end
+
+      state :comment do
+        rule %r/\*\)/, Comment::Multiline, :pop!
+        rule %r/\(\*/, Comment::Multiline, :comment
+        rule %r/[^*)]+/, Comment::Multiline
+        rule %r/[*)]/, Comment::Multiline
+      end
+
+      state :root do
+        mixin :basic
+
+        rule %r/\w[\w']*/ do |m|
+          name = m[0]
+          if name == "module"
+            token Keyword::Reserved
+            push :module
+          elsif self.class.reserved.include? name
+            token Keyword::Reserved
+          elsif self.class.types.include? name
+            token Keyword::Type
+          elsif name =~ /\A[A-Z]/
+            token Keyword::Namespace
+          else
+            token Name
+          end
+        end
+
+        rule %r/"/, Str, :string
+        rule %r/\//, Str, :regexp
+
+        rule %r([-*+.=?\|]+), Operator
+        rule %r/[\[\](){}:]/, Punctuation
+      end
+
+      state :module do
+        rule %r/\s+/, Text
+        # module Foo (functions)
```

Comment:
I didn't see this usage in the Augeas documentation. So you can write `module Foo (functions)`? What is `functions` in that case? It will be highlighted by this state as a namespace.

Reply:
No actually, you're right. This is probably left from bits I copied from the haskell lexer.